kedro/framework/cli/catalog.py

```diff
@@ -265,9 +265,14 @@ def resolve_patterns(metadata: ProjectMetadata, env: str) -> None:
                 ds_name, matched_pattern, ds_config_copy
             )
 
-            ds_config["filepath"] = _trim_filepath(
-                str(context.project_path) + "/", ds_config["filepath"]
-            )
+            path_keys_to_check = ["filepath", "path"]
+
+            for key in path_keys_to_check:
+                if key in ds_config:
+                    ds_config[key] = _trim_filepath(
+                        str(context.project_path) + "/", ds_config[key]
+                    )
+
```

Comment:
@AhdraMeraliQB @merelcht  Could we get rid of this whole check instead of hard-coding this? 
I see this was a design decision (https://github.com/kedro-org/kedro/pull/2891#issuecomment-1665306293)
I think this was before the code went through a few changes. Even for explicit datasets, we just get the config as it is written in the catalog using the config loader so relative path instead of the absolute path so everything is uniform even without this check.

Reply:
Works for me - tried removing this check locally and it all the tests still pass.

Reply:
Ah good point @ankatiyar. If everything still works as expected and the problem with `PartitionedDataset` is solved when the check is removed, then that definitely seems like the best solution.

Reply:
in that case, @MosaicMan could you get rid of these lines entirely and the `_trim_filepath()` method below? And add this change to the release notes as well! :)